packages/itwinui-react/src/core/LabeledTextarea/LabeledTextarea.tsx

```diff
@@ -56,48 +67,69 @@ export const LabeledTextarea = React.forwardRef((props, ref) => {
   const uid = useId();
 
   const {
-    className,
-    style,
     disabled = false,
     label,
     message,
     status,
-    textareaClassName,
-    textareaStyle,
     displayStyle = 'default',
     iconDisplayStyle = displayStyle === 'default' ? 'block' : 'inline',
     svgIcon,
     required = false,
     id = uid,
+    wrapperProps,
+    labelProps,
+    messageContentProps,
+    iconProps,
     ...textareaProps
   } = props;
 
   const icon = svgIcon ?? (status && StatusIconMap[status]());
+  const iconFill = !svgIcon ? status : undefined;
 
   return (
-    <InputContainer
-      label={label}
-      disabled={disabled}
-      required={required}
-      status={status}
-      message={message}
-      icon={icon}
-      isLabelInline={displayStyle === 'inline'}
-      isIconInline={iconDisplayStyle === 'inline'}
-      className={className}
-      style={style}
-      inputId={id}
-    >
-      <Textarea
-        disabled={disabled}
-        className={textareaClassName}
-        style={textareaStyle}
-        required={required}
-        id={id}
-        {...textareaProps}
-        ref={ref}
-      />
-    </InputContainer>
+    <InputGrid labelPlacement={displayStyle} {...wrapperProps}>
```

Comment:
LabeledTextarea could reuse LabeledInput? it would greatly simplify and avoid duplicating most of this code.

```tsx
export const LabeledTextarea = React.forwardRef((props, forwardedRef) => {
  return (
    // ref types don't match but it's internal ts-ignore is ok here
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    <LabeledInput as='textarea' rows={3} ref={forwardedRef} {...props} />
  );
}) as PolymorphicForwardRefComponent<'textarea', LabeledTextareaProps>;
```

Reply:
Changed